src/hackney_pool.erl

```diff
@@ -304,6 +307,12 @@ handle_cast({set_maxconn, MaxConn}, State) ->
 handle_cast({set_timeout, NewTimeout}, State) ->
   {noreply, State#state{timeout=NewTimeout}};
 
+handle_cast({checkout_cancel, Dest, RequestRef}, State) ->
```

Comment:
How much should it be different from a 'DOWN' case ?
https://github.com/benoitc/hackney/blob/master/src/hackney_pool.erl

imo we should go back in the queue and trigger another waiting client in such case. Thoughts?

Reply:
Sounds like a good idea to me, reworking...